pennylane/transforms/core/transform_program.py

```diff
@@ -276,36 +281,200 @@ def has_final_transform(self) -> bool:
         """Check if the transform program has a terminal transform or not."""
         return self[-1].final_transform if self else False
 
+    def has_classical_cotransform(self) -> bool:
+        """Check if the transform program has some classical cotransforms.
+
+        Returns:
+            bool: Boolean
+        """
+        return any(t.classical_cotransform is not None for t in self)
+
+    def _set_all_classical_jacobians(
```

Comment:
Is there any expected use case for classical cotransforms apart from gradients? I'm asking because all of the code in `TransformProgram` pertaining to classical cotransforms is gradient/jacobian specific, and it should be more general if there are non-gradient uses for classical cotransforms

Reply:
Yes the classical co-transform has the only goal to deal with classical gradients and is not intended to be used for other purposes.